compendium/DeclarativeServices/src/SCRBundleExtension.cpp

```diff
@@ -76,8 +86,55 @@ SCRBundleExtension::SCRBundleExtension(const cppmicroservices::BundleContext& bu
                   std::current_exception());
     }
   }
+logger->Log(cppmicroservices::logservice::SeverityLevel::LOG_DEBUG,
+            "Created instance of SCRBundleExtension for " +
+                bundleContext.GetBundle().GetSymbolicName());
+
+// Add a configuration object to ConfigAdmin for each configuration in the cm metadata.
+const auto& headers = bundleContext.GetBundle().GetHeaders();
+if (headers.count(cppmicroservices::util::CMConstants::CM_KEY) == 0u) {
+    logger->Log(cppmicroservices::logservice::SeverityLevel::LOG_DEBUG,
+                "No cm components found in bundle " +
+                bundleContext.GetBundle().GetSymbolicName());
+    return;
+}
+try {   
+    auto const& cmMetadata =
+        cppmicroservices::ref_any_cast<cppmicroservices::AnyMap>(
+        headers.at(cppmicroservices::util::CMConstants::CM_KEY));
+
+    auto cmVersion =
+        cppmicroservices::any_cast<int>(cmMetadata.at(cppmicroservices::util::CMConstants::CM_VERSION));
+    auto metadataParser =
+        cppmicroservices::util::MetadataParserFactory::Create(cmVersion, logger);
+    auto configurationMetadata =
+        metadataParser->ParseAndGetConfigurationMetadata(cmMetadata);
+
+    auto sr = this->bundleContext.GetServiceReference<
+        cppmicroservices::service::cm::ConfigurationAdmin>();
+    auto configAdmin =
+        this->bundleContext
+        .GetService<cppmicroservices::service::cm::ConfigurationAdmin>(sr);
+
+    for (auto& oneCmMetadata : configurationMetadata) {
```

Comment:
make this `const auto& oneCmMetadata` if possible.

Reply:
Fixed